nss/integration-tests/integration_test.go

```diff
@@ -153,23 +156,30 @@ func TestIntegration(t *testing.T) {
 }
 
 func TestMain(m *testing.M) {
-	// Build the NSS library and executable in a temporary directory and allow linking to it.
-	tmpDir, cleanup, err := createTempDir()
-	if err != nil {
-		os.Exit(1)
-	}
-	defer cleanup()
-
-	targetDir, libPath = tmpDir, filepath.Join(tmpDir, "libnss_aad.so.2")
-	// Builds the NSS Library.
-	if err = buildRustNSSLib(); err != nil {
-		cleanup()
-		fmt.Fprintf(os.Stderr, "Can not build rust nss library: %v", err)
-		os.Exit(1)
-	}
-
 	testutils.InstallUpdateFlag()
 	flag.Parse()
 
-	m.Run()
+	code := m.Run()
+	defer os.Exit(code)
+
+	if err := testutils.MergeCoverages(); err != nil {
+		log.Printf("Teardown: failed to merge coverage files: %v", err)
+
+		// This ensures that we fail the test if we can't merge the coverage files, if the test
+		// was successfull, otherwise we exit with the code returned by m.Run()
+		if code == 0 {
+			defer os.Exit(24)
+		}
+	}
+
+	if testutils.WantCoverage() {
+		// Gets the path to the integration-tests.
+		_, p, _, _ := runtime.Caller(0)
+		currDir := filepath.Dir(p)
+		// Purge .profraw files
+		files, _ := filepath.Glob(filepath.Join(currDir, "*.profraw"))
+		for _, f := range files {
+			_ = os.Remove(f)
+		}
+	}
```

Comment:
Since the defer calls go on a stack, I had to update this scope as well to ensure that the files will be cleaned correctly

Reply:
Just to check: the .profraw is in a subdirectory of the current repo? (I would prefer tempdir if possible, but otherwise, should be in `target/` so that we can avoid committing them by accident if something bad happens)

Reply:
Right now they are being saved in the integration-tests directory (I'm not sure why)